notification-hubs-sdk/src/test/java/com/microsoft/windowsazure/messaging/notificationhubs/TagEnricherTest.java

```diff
@@ -0,0 +1,79 @@
+package com.microsoft.windowsazure.messaging.notificationhubs;
+
+import android.content.Context;
+
+import androidx.test.core.app.ApplicationProvider;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.robolectric.RobolectricTestRunner;
+import org.robolectric.annotation.Config;
+
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static org.junit.Assert.*;
+
+@RunWith(RobolectricTestRunner.class)
+@Config(sdk = 28)
+public class TagEnricherTest {
+    private Context context = ApplicationProvider.getApplicationContext();
+    private List<String> tagList = Stream.of("tag1", "tag2", "tag3").collect(Collectors.toList());
+
+    @Test
+    public void TagEnricherAddTag () {
+        TagEnricher te = new TagEnricher();
+        te.SetPreferences(context);
+
+        assertTrue(te.addTag(tagList.get(0)));
+        assertTrue(((HashSet<String>)te.getTags()).contains(tagList.get(0)));
+    }
+
+    @Test
+    public void TagEnricherAddTags () {
+        TagEnricher te = new TagEnricher();
+        te.SetPreferences(context);
+
+        assertTrue(te.addTags(tagList));
+        assertTrue(((HashSet<String>)te.getTags()).containsAll(tagList));
+        assertEquals(((HashSet<String>)te.getTags()).size(), tagList.size());
```

Comment:
I'd update this test to include adding a second set of tags, and ensuring that both sets of tags are in the resulting collection.